lib/datadog/core/remote/client.rb

```diff
@@ -185,8 +194,35 @@ def capabilities_binary
         def register_receivers
           matcher = Dispatcher::Matcher::Product.new(products)
 
-          dispatcher.receivers << Dispatcher::Receiver.new(matcher) do |_repository, changes|
-            changes.each { |change| Datadog.logger.debug { "remote config change: #{change.path.inspect}" } }
+          dispatcher.receivers << Dispatcher::Receiver.new(matcher) do |repository, changes|
+            changes.each do |change|
+              Datadog.logger.debug { "remote config change: '#{change.path}'" }
+            end
+
+            rules = repository.contents.select do |content|
+              content.path.product == 'ASM_DD'
+            end.map! { |content| JSON.parse(content.data.read.tap { content.data.rewind }) }
+
+            data = repository.contents.select do |content|
+              content.path.product == 'ASM_DATA' && ['blocked_ips', 'blocked_users' ].include?(content.path.config_id)
+            end.map! { |content| JSON.parse(content.data.read.tap { content.data.rewind }) }
+
+            overrides = repository.contents.select do |content|
+              content.path.product == 'ASM' && ['blocking', 'disabled_rules'].include?(content.path.config_id)
+            end.map! { |content| JSON.parse(content.data.read.tap { content.data.rewind }) }
+
+            exclusions = repository.contents.select do |content|
+              content.path.product == 'ASM' && content.path.config_id == 'exclusion_filters'
+            end.map! { |content| JSON.parse(content.data.read.tap { content.data.rewind }) }
+
+            ruleset = AppSec::Processor::RuleMerger.merge(
+              rules: rules,
```

Comment:
Is. Is it a warranty that rules will always be populated here? 

If the remote config does not send the `ASM_DD` product information, the `RuleMerger` might break.